code/game/machinery/washing_machine.dm

```diff
@@ -188,101 +186,95 @@
 		QDEL_NULL(crayon)
 
 
-	if(locate(/mob,contents))
-		state = 7
-		gibs_ready = 1
+	if(locate(/mob) in src)
+		state = CLOSED_BLOODY
+		gibs_ready = TRUE
 	else
-		state = 4
+		state = CLOSED_FULL
 	update_icon(UPDATE_ICON_STATE)
 
 /obj/machinery/washing_machine/update_icon_state()
 	icon_state = "wm_[state][panel]"
 
-/obj/machinery/washing_machine/attackby(obj/item/W as obj, mob/user as mob, params)
+/obj/machinery/washing_machine/attackby(obj/item/W, mob/user, params)
 	if(default_unfasten_wrench(user, W))
 		return
-	if(istype(W,/obj/item/toy/crayon) ||istype(W,/obj/item/stamp))
-		if(state in list(	1, 3, 6))
+	if(istype(W, /obj/item/toy/crayon) || istype(W, /obj/item/stamp))
+		if(state in list(OPEN_EMPTY, OPEN_FULL))
 			if(!crayon)
 				user.drop_item()
 				crayon = W
-				crayon.loc = src
+				crayon.forceMove(src)
 				update_icon(UPDATE_ICON_STATE)
 			else
 				return ..()
 		else
 			return ..()
-	else if(istype(W,/obj/item/grab))
-		if((state == 1) && hacked)
+	else if(istype(W, /obj/item/grab))
+		if(state == OPEN_EMPTY)
 			var/obj/item/grab/G = W
 			if(ishuman(G.assailant) && iscorgi(G.affecting))
-				G.affecting.loc = src
+				G.affecting.forceMove(src)
 				qdel(G)
-				state = 3
+				state = OPEN_FULL
 			update_icon(UPDATE_ICON_STATE)
 		else
 			return ..()
-	else if(istype(W,/obj/item/stack/sheet/hairlesshide) || \
-		istype(W,/obj/item/clothing/under) || \
-		istype(W,/obj/item/clothing/mask) || \
-		istype(W,/obj/item/clothing/head) || \
-		istype(W,/obj/item/clothing/gloves) || \
-		istype(W,/obj/item/clothing/shoes) || \
-		istype(W,/obj/item/clothing/suit) || \
-		istype(W,/obj/item/bedsheet))
+	else if(istype(W, /obj/item/stack/sheet/hairlesshide) || \
+		istype(W, /obj/item/clothing/under) || \
+		istype(W, /obj/item/clothing/mask) || \
+		istype(W, /obj/item/clothing/head) || \
+		istype(W, /obj/item/clothing/gloves) || \
+		istype(W, /obj/item/clothing/shoes) || \
+		istype(W, /obj/item/clothing/suit) || \
+		istype(W, /obj/item/bedsheet))
```

Comment:
If you're cleaning most of the code anyways, could you do this with a typecache or something?